### PR TITLE
➕ add ML books and courses

### PR DESCRIPTION
**Pull request by yashppawar**
## What does this PR do?
- Add resource(s) 
- Add info

### Description
- Add [The Mechanics of Machine Learning](https://mlbook.explained.ai/) book
- Add [Made with ML](https://madewithml.com/) course
- Add Mathematics for Machine Learning By Imperial College London on YouTube
- Change URL for Introduction to Statistical Learning to a more direct URL
- Change URL for Mining of Massive Datasets to a more direct URL 

### Why is this valuable (or not)?
The added resources help to learn Machine Learning, MLOps.
The changed URL makes it one click to access the pdf.

### How do we know it's really free?
The resources do not ask for any email/credit card info to get access and the site is either a trusted site or an official site.

### For book lists, is it a book? For course lists, is it a course? etc.
* Books
    - The Mechanics of Machine Learning
    - Introduction to Statistical Learning 
    - Mining of Massive Datasets
 * Courses
   - Mathematics for Machine Learning (Video)
   - Made With ML (misc)

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by yashppawar** [comment]
ok, all checks passed now 😌

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1903,7 +1903,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 ### R
 
 * [Advanced R Programming](http://adv-r.had.co.nz) - Hadley Wickham
-* [An Introduction to Statistical Learning with Applications in R](http://www-bcf.usc.edu/~gareth/ISL/) - Gareth James, Daniela Witten, Trevor Hastie and Robert Tibshirani (PDF)
+* [An Introduction to Statistical Learning with Applications in R](https://web.stanford.edu/~hastie/ISLR2/ISLRv2_website.pdf) - Gareth James, Daniela Witten, Trevor Hastie and Robert Tibshirani [website](https://www.statlearning.com) (PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1906]: no links for authors

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -213,7 +213,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Internet Advertising: An Interplay among Advertisers, Online Publishers, Ad Exchanges and Web Users](http://arxiv.org/pdf/1206.1754v2.pdf) (PDF)
 * [Introduction to Data Science](https://docs.google.com/file/d/0B6iefdnF22XQeVZDSkxjZ0Z5VUE/edit?pli=1) - Jeffrey Stanton
 * [Introduction to Data Science](https://leanpub.com/datasciencebook) - Rafael A Irizarry *(Leanpub account or valid email requested)*
-* [Mining of Massive Datasets](http://www.mmds.org)
+* [Mining of Massive Datasets](http://infolab.stanford.edu/~ullman/mmds/book.pdf) - Jure Leskovec, Anand Rajaraman, Jeffrey D. Ullman [website](https://www.mmds.org) (PDF)
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:216]: no author links please

File: books/free-programming-books-subjects.md [hunk 3]
```diff
@@ -283,7 +283,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [A First Encounter with Machine Learning](https://www.ics.uci.edu/~welling/teaching/ICS273Afall11/IntroMLBook.pdf) (PDF)
 * [A Selective Overview of Deep Learning](https://arxiv.org/abs/1904.05526) - Fan, Ma, and Zhong (PDF)
 * [Algorithms for Reinforcement Learning](https://sites.ualberta.ca/~szepesva/papers/RLAlgsInMDPs.pdf) - Csaba Szepesvári (PDF)
-* [An Introduction to Statistical Learning](http://www-bcf.usc.edu/~gareth/ISL/) - Gareth James, Daniela Witten, Trevor Hastie and Robert Tibshirani
+* [An Introduction to Statistical Learning](https://web.stanford.edu/~hastie/ISLR2/ISLRv2_website.pdf) - Gareth James, Daniela Witten, Trevor Hastie and Robert Tibshirani [website](https://www.statlearning.com) (PDF)
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:286]: no author links please

File: courses/free-courses-en.md [hunk 4]
```diff
@@ -545,6 +545,8 @@
 * [Machine Learning Recipes with Josh Gordon](https://www.youtube.com/playlist?list=PLOU2XLYxmsIIuiBfYad6rFYQU_jL2ryal)
 * [Machine Learning Tutorial Python \| Machine Learning For Beginners](https://www.youtube.com/playlist?list=PLeo1K3hjS3uvCeTYTeyfe0-rN5r8zn9rw) - Dhaval Patel
 * [Machine Learning with Python by Saeed Aghabozorgi](https://cognitiveclass.ai/courses/machine-learning-with-python) (cognitiveclass.ai)
+* [Mathematics for Machine Learning - Linear Algebra](https://www.youtube.com/playlist?list=PLiiljHvN6z1_o1ztXTKWPrShrMrBLo5P3)
```
Comment (eshellman) [inline courses/free-courses-en.md:548]: creator?

File: courses/free-courses-en.md [hunk 5]
```diff
@@ -545,6 +545,8 @@
...
+* [Mathematics for Machine Learning - Multivariate Calclus](https://www.youtube.com/playlist?list=PLiiljHvN6z193BBzS0Ln8NnqQmzimTW23)
```
Comment (eshellman) [inline courses/free-courses-en.md:549]: creator?

**Review by eshellman — CHANGES_REQUESTED** [review]

**Comment by yashppawar** [comment]
Removed links, and added author to MML course.

File: courses/free-courses-en.md (hunk 4 above)
Comment (yashppawar) [reply courses/free-courses-en.md]: Added in the latest commit

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**